crates/sui/src/fire_drill.rs

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A tool to semi automate fire drills. It still requires some manual work today. For example,
+//! 1. update iptables for new tpc/udp ports
+//! 2. restart the node in a new epoch when config file will be reloaded and take effects
+//!
+//! Example usage:
+//! sui fire-drill metadata-rotation \
+//! --sui-node-config-path validator.yaml \
+//! --fullnode-rpc-url http://fullnode-my-local-net:9000
+
+use anyhow::bail;
+use clap::*;
+use fastcrypto::ed25519::Ed25519KeyPair;
+use fastcrypto::traits::{KeyPair, ToFromBytes};
+use move_core_types::ident_str;
+use multiaddr::Multiaddr;
+use std::path::{Path, PathBuf};
+use sui_config::node::KeyPairWithPath;
+use sui_config::utils;
+use sui_config::{node::AuthorityKeyPairWithPath, Config, NodeConfig, PersistedConfig};
+use sui_json_rpc_types::{SuiExecutionStatus, SuiTransactionResponseOptions};
+use sui_sdk::{rpc_types::SuiTransactionEffectsAPI, SuiClient, SuiClientBuilder};
+use sui_types::base_types::{ObjectRef, SuiAddress};
+use sui_types::crypto::{generate_proof_of_possession, get_key_pair, SuiKeyPair};
+use sui_types::messages::{CallArg, ObjectArg, TransactionData};
+use sui_types::utils::to_sender_signed_transaction;
+use sui_types::{committee::EpochId, crypto::get_authority_key_pair};
+use sui_types::{
+    SUI_FRAMEWORK_OBJECT_ID, SUI_SYSTEM_STATE_OBJECT_ID, SUI_SYSTEM_STATE_OBJECT_SHARED_VERSION,
+};
+use tracing::info;
+
+#[derive(Parser)]
+pub enum FireDrill {
```

Comment:
Is this tool just for fire drill or actually needed by validator admins for operations?

Reply:
It's often for fire drills. I'm working on a validator tool atm